### PR TITLE
MIC-146 retrieve csip record by uuid

### PR DESCRIPTION
**Pull request by anthony-britton-moj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/RetrieveCsipRecordIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  @ParameterizedTest
+  @ValueSource(strings = [ROLE_CSIP_UI, ROLE_NOMIS])
+  fun `200 ok - returns matching CSIP record`(role: String) {
+    val prisonNumber = givenValidPrisonNumber("G1234CR")
+    val record = givenCsipRecordWithReferral(generateCsipRecord(prisonNumber))
+
+    val response = getCsipRecord(record.recordUuid, role)
+    with(response) {
+      assertThat(prisonNumber).isEqualTo(prisonNumber)
```
Comment (JackReeveMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/RetrieveCsipRecordIntTest.kt:60]: Assuming the first `prisonNumber` here comes from `response` and the second is the variable on line 55, is kotlin smart enough to know that you're not referencing the same variable here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/InvestigationsIntTest.kt [hunk 2]
```diff
@@ -87,7 +87,7 @@ class InvestigationsIntTest : IntegrationTestBase() {
   fun `400 bad request - invalid Outcome Type code`() {
     val recordUuid = UUID.randomUUID()
     val request = investigationRequest(interviewRequest(roleCode = "WRONG_CODE"))
-    val response = createInvestigationResponseSpec(recordUuid, request).badRequest()
+    val response = createInvestigationResponseSpec(recordUuid, request).errorResponse(HttpStatus.BAD_REQUEST)
```
Comment (JackReeveMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/InvestigationsIntTest.kt:90]: What's the benefit to changing all of these away from `.badRequest()` ?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/RetrieveCsipRecordIntTest.kt (hunk 1 above)
Comment (iterative-it) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/resource/RetrieveCsipRecordIntTest.kt:60]: Good spot @JackReeveMoJ. It does need changing to `assertThat(this.prisonNumber).isEqualTo(prisonNumber)` to verify prison number from the response.

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**